app/models/message.rb

```diff
@@ -10,7 +10,7 @@ class Message < ApplicationRecord
 
   before_validation :set_thread_id, :remove_deleted_recipient
   before_create :check_recipient
-  after_create :notify_recipient
+  after_create_commit :notify_recipient
```

Comment:
Did we figure out if `deliver` aliases to `now` or `later`?

Reply:
I don't think so, but either way we want this on an after_commit?

Reply:
It looks like `deliver` comes from `resque_mailer`: https://github.com/zapnap/resque_mailer/blob/35c2a7c7ff9038b9bff95c50190d162f95afcd5f/lib/resque_mailer.rb#L137

It defaults to a behavior like `deliver_later` if it can do that through resque, or else falls back to synchronous (like `deliver_now`), but really isn't either of them! Maybe we should investigate using the ActionMailer native jobs?